frontend/src/components/UserForm/UserForm.js

```diff
@@ -0,0 +1,198 @@
+import { useMutate } from "../../hooks/use-mutate"
+import { useRef, useState, useEffect} from "react"
+import { useUser } from "../../hooks/user-hooks"
+import { useParams } from "react-router"
+import { CountryDropdown, RegionDropdown } from "react-country-region-selector";
+
+const UserForm = () => {
+  const formRef = useRef()
+  const { updateUser } = useMutate()
+  const { id } = useParams()
+  const user = useUser({ "id": id })
+  const isAdmin = user?.data?.data?.data.user.admin
+
+  const fetchedCountry = user?.data?.data?.data?.user.country;
+  const fetchedState = user?.data?.data?.data?.user.state;
```

Comment:
I know that we've typically done this in the past, but due to the amount of items on this form, this method looks pretty ugly. Destructuring the data looks to be the best way to keep this nice and clean. We should destructure the return for the user instead of grabbing the entire object like that.

I would refactor this like so:
```
  const { status, data: userData } = useUser({ id })

  const user = userData?.data?.data?.user
  const isAdmin = user?.admin

  const fetchedCountry = user?.country
  const fetchedState = user?.state
```

Reply:
refactored, looks much cleaner now!